pyriemann/utils/mean.py

```diff
@@ -304,6 +305,56 @@ def mean_ale(covmats, tol=10e-7, maxiter=50, sample_weight=None):
     return C
 
 
+def mean_alm(covmats, tol=1e-14, maxiter=1000,
+             verbose=False, sample_weight=None):
+    r"""Return Ando-Li-Mathias (ALM) mean
+
+    Find the geometric mean recursively [1], generalizing from:
+    
+    .. math::
+        \mathbf{C} = A^{\frac{1}{2}}(A^{-\frac{1}{2}}B^{\frac{1}{2}}A^{-\frac{1}{2}})^{\frac{1}{2}}A^{\frac{1}{2}}
+
+    require a high number of iterations.
+
+    This is the adaptation of the Matlab code proposed by Dario Bini and
+    Bruno Iannazzo, http://bezout.dm.unipi.it/software/mmtoolbox/
+    Extremely slow, due to the recursive formulation.
+
+    :param covmats: Covariance matrices set, (n_trials, n_channels, n_channels)
+    :param tol: the tolerance to stop the gradient descent
+    :param maxiter: maximum number of iteration, default 100
+    :param verbose: indicate when reaching maxiter
+    :param sample_weight: the weight of each sample
+
+    :returns: Karcher mean covariance matrix
+
+    References
+    ----------
+    [1] T. Ando, C.-K. Li and R. Mathias, "Geometric Means", Linear Algebra
+        Appl. 385 (2004), 305-334.
+    """
+    sample_weight = _get_sample_weight(sample_weight, covmats)
```

Comment:
Major problem : in the current implementation, weights are totally unused.

Reply:
Right! It is corrected, I added weights in the computation.

Reply:
Great! Can you add weights in tests too?